airflow-core/src/airflow/utils/db.py

```diff
@@ -1638,9 +1639,9 @@ def __getitem__(self, key: slice) -> Sequence[T]: ...
     def __getitem__(self, key: int | slice) -> T | Sequence[T]:
         if isinstance(key, int):
             if key >= 0:
-                stmt = self._select_asc.offset(key)
+                stmt = cast("Select", self._select_asc).offset(key)
             else:
-                stmt = self._select_desc.offset(-1 - key)
+                stmt = cast("Select", self._select_desc).offset(-1 - key)
```

Comment:
I don’t think doing this cast repeatedly is a good idea. `_select_asc` and `_select_desc` should keep their current annotation, and cast should just be done when those values are set.